lib/imapoptions

```diff
@@ -1206,15 +1206,7 @@ Blank lines and lines beginning with ``#'' are ignored.
    is prepended. */
 
 { "improved_mboxlist_sort", 0, SWITCH, "2.3.17" }
```

Comment:
Also need to mechanically mark the option as deprecated, like this:

```suggestion
{ "improved_mboxlist_sort", 0, SWITCH, "UNRELEASED", "UNRELEASED" }
```

The first occurrence of "UNRELEASED" here replaces the "2.3.17" as being the most recent version in which something about this option changed (the fact it is being deprecated is a change).  The second occurrence marks the option as being deprecated (as of the specified version).  The magic word "UNRELEASED" is how we refer to a release that doesn't yet exist without trying to predict what its version number will be.  It'll be replaced with a real version number when a real version is released.